cli/azd/cmd/up.go

```diff
@@ -52,121 +43,67 @@ func newUpCmd() *cobra.Command {
 	runner middleware.MiddlewareContext,
-	prompters prompt.Prompter,
 	provisioningManager *provisioning.Manager,
 ) actions.Action {
 	return &upAction{
-		flags:                      flags,
-		env:                        env,
-		projectConfig:              projectConfig,
-		packageActionInitializer:   packageActionInitializer,
-		provisionActionInitializer: provisionActionInitializer,
-		deployActionInitializer:    deployActionInitializer,
-		console:                    console,
-		runner:                     runner,
-		prompters:                  prompters,
-		provisioningManager:        provisioningManager,
+		flags:                        flags,
+		console:                      console,
+		env:                          env,
+		projectConfig:                projectConfig,
+		workflowRunActionInitializer: workflowRunActionInitializer,
+		runner:                       runner,
+		provisioningManager:          provisioningManager,
 	}
 }
 
 func (u *upAction) Run(ctx context.Context) (*actions.ActionResult, error) {
-	if u.flags.provisionFlags.noProgress {
-		fmt.Fprintln(
-			u.console.Handles().Stderr,
-			//nolint:lll
-			output.WithWarningFormat(
-				"WARNING: The '--no-progress' flag is deprecated and will be removed in a future release.",
-			),
-		)
-		// this flag actually isn't used by the provision command, we set it to false to hide the extra warning
-		u.flags.provisionFlags.noProgress = false
-	}
-
-	if u.flags.deployFlags.serviceName != "" {
-		fmt.Fprintln(
-			u.console.Handles().Stderr,
-			//nolint:lll
-			output.WithWarningFormat("WARNING: The '--service' flag is deprecated and will be removed in a future release."))
-	}
-
-	err := u.provisioningManager.Initialize(ctx, u.projectConfig.Path, u.projectConfig.Infra)
-	if err != nil {
-		return nil, err
-	}
-
-	startTime := time.Now()
-
-	packageAction, err := u.packageActionInitializer()
-	if err != nil {
-		return nil, err
-	}
-	packageOptions := &middleware.Options{CommandPath: "package"}
-	_, err = u.runner.RunChildAction(ctx, packageOptions, packageAction)
-	if err != nil {
-		return nil, err
+	if u.projectConfig.Workflows == nil {
+		u.projectConfig.Workflows = map[string][]*workflow.Step{}
 	}
 
-	provision, err := u.provisionActionInitializer()
-	if err != nil {
-		return nil, err
-	}
+	upWorkflow, has := u.projectConfig.Workflows["up"]
+	if !has || len(upWorkflow) == 0 {
+		err := u.provisioningManager.Initialize(ctx, u.projectConfig.Path, u.projectConfig.Infra)
+		if err != nil {
+			return nil, err
+		}
 
-	provision.flags = &u.flags.provisionFlags
-	provisionOptions := &middleware.Options{CommandPath: "provision"}
-	provisionResult, err := u.runner.RunChildAction(ctx, provisionOptions, provision)
-	if err != nil {
-		return nil, err
+		u.projectConfig.Workflows["up"] = defaultUpWorkflow
+	} else {
+		u.console.Message(ctx, output.WithWarningFormat("WARNING: Running custom 'up' workflow from azure.yaml"))
```

Comment:
I do not think we should warn here.  If someone needs to change what `up` means, hitting them with a warning every time they run `up` for the rest of time is not going to delight them.  If you are concerned about people not knowing what is going on, I'd be fine with a log statement so --debug shows what is going on.

Reply:
Perhaps `warning` is not the best choice here to be so upfront in your face.  Maybe just a greyed out note?  I do feel that some visual indication is helpful to highlight that `azd up` is not doing its default thing but rather running a custom workflow.